sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/collect.scala

```diff
@@ -46,13 +46,15 @@ abstract class Collect[T <: Growable[Any] with Iterable[Any]] extends TypedImper
   // actual order of input rows.
   override lazy val deterministic: Boolean = false
 
+  def getValueOnUpdate(value: Any): Any = InternalRow.copyValue(value)
```

Comment:
Make this protected. Can we name this `converToBufferElement` or something like that.